en/components/circular_progress.md

```diff
@@ -9,19 +9,107 @@ _keywords: Ignite UI for Angular, UI controls, Angular widgets, web widgets, UI
+...
+import { IgxProgressBarModule } from 'igniteui-angular/main';
+
+@NgModule({
+    ...
+    imports: [..., IgxProgressBarModule],
+    ...
+})
+export class AppModule {}
+```
+And now to have a better understanding how everything works, let's create a simple example, in which we will simulate a real process progress, that is
+triggered on button click. In order to make out example better we will need to import and some additional modules in the **app.module.ts** file.
+
+```typescript
+// app.module.ts
+import {
+    ..., IgxButtonModule, IgxIconModule, IgxRippleModule
+} from 'igniteui-angular/main';
+
+@NgModule({
+    ...
+    imports: [..., IgxButtonModule, IgxIconModule, IgxRippleModule],
+})
+export class AppModule {}
+```
+Notice that the **igx-circular-bar** emits `onProgressChanged` event that outputs object that give us `{currentValue: 65, previousValue: 64}` on each step.
+
 ```html
-<igx-circular-bar (onProgressChanged)="f($event)" [value]="currentValue">
-</igx-circular-bar>
+<section class="sample-content">
+    <article class="sample-column">
+      <h5>Text is hidden</h5>
+      <div class="circular-container">
+        <igx-circular-bar [value]="currentValue" [max]="100" [animate]="true" [textVisibility]="false" (onProgressChanged)="progresChanged($event)"></igx-circular-bar>
+      </div>
+    </article>
+    <article class="sample-column">
+      <h5>Text is displayed</h5>
+      <div class="circular-container">
+        <igx-circular-bar [value]="currentValue" [max]="100" [animate]="true" [textVisibility]="true" (onProgressChanged)="progresChanged($event)"></igx-circular-bar>
+      </div>
+     </article>
+</section>
+<div class="button-container">
+    <p>Press the button to start updating the bar</p>
+    <button igxButton="fab" igxButtonBackground="#333" igxRipple="white" (click)="tick()">
+        <igx-icon fontSet="material" [name]="changeIcon()"></igx-icon>
+    </button>
+</div>
 ```
+
+```typescript
+...
+  public currentValue: number;
+  public interval: any;
+  @ViewChild(IgxCircularProgressBarComponent) public circularBar: IgxCircularProgressBarComponent;
+  public ngOnInit() {
+    this.currentValue = 0;
+  }
+  public changeIcon() {
+    return this.interval ? "pause" : "play_arrow";
+  }
+  public tick() {
+    if (this.interval) {
+        this.interval = clearInterval(this.interval);
+        return;
+    }
+    this.interval = setInterval(this.updateValue.bind(this), 60);
+  }
+  public updateValue() {
+     this.circularBar.updateProgressSmoothly(this.currentValue += this.randomIntFromInterval(1, 3), 1);
+     if (this.circularBar.value > this.circularBar.max + 3) {
+       this.interval = clearInterval(this.interval);
+     }
+  }
+  public progresChanged(progress) { ... }
+  private randomIntFromInterval(min: number, max: number) {
+    return Math.floor(Math.random() * (max - min + 1) + min);
+  }
+```
+
+And now if we set up everything correct we should have the folllowing displayed in our browser:
```

Comment:
if we set up everything correctly
What I wrote is grammatically correct, although enough native English speakers don't know that and would be ok with what you wrote. :)

Reply:
👍